py-polars/polars/expr/array.py

```diff
@@ -15,6 +16,9 @@
 class ExprArrayNameSpace:
     """Namespace for array related expressions."""
 
+    if not BUILDING_SPHINX_DOCS:
+        __slots__ = ("_pyexpr",)
```

Comment:
I'm not a fan of this construction. I would expect the `sphinx_accessor` to take care of this, but it doesn't. We should fix that directly rather than introducing this check everywhere.

Reply:
Agree on this. Will try to look into this.
I've tried monkey-patching, but dynamically  adding `__dict__` is way harder when `__slots__` are in place, because it modifies object layout.
Problem seems to be that autosummary makes objects into "modules" and sphinx expects module objects to have dict, so exception thrown from sphinx itself.
Maybe will try to make subclass with `__dict__` dynamically.

Reply:
Created pull request to handle this https://github.com/xarray-contrib/sphinx-autosummary-accessors/pull/124

Reply:
That's great! They have not released in a year though, so I don't expect this to be picked up quickly.

I'd suggest removing the namespace slots for now - we can add them again when the issue is resolved.